src/components/ShareStructure/EditShareStructure.vue

```diff
@@ -280,7 +280,10 @@ export default class EditShareStructure extends Mixins(CurrencyLookupMixin) {
     let rules: Array<Function> = []
     if (!this.hasNoMaximumShares) {
       rules = [
-        (v: string) => /^\d+$/.test(v) || 'Must be a number greater than 0']
+        (v: string) => (v !== null && v !== '') || 'Number of shares is required',
+        (v: string) => /^-?\d+$/.test(v) || 'Must be a whole number',
+        (v: string) => +v > 0 || 'Number must be greater than 0'
```

Comment:
Sorry remind me again, `+` is a type casting to a number?

Reply:
Yes!